src/extensions/client/endpoint.rs

```diff
@@ -38,19 +40,23 @@ impl Endpoint {
         health_config: HealthCheckConfig,
     ) -> Self {
         let (client_tx, client_rx) = tokio::sync::watch::channel(None);
+        let (reconnect_tx, mut reconnect_rx) = tokio::sync::mpsc::channel(1);
```

Comment:
`tokio::sync::Notify` may be a better option

Reply:
notify is one off thing but we may need to reconnect multiple times